app/(auth)/signup/page.tsx

```diff
@@ -0,0 +1,131 @@
+    const newPassword = e.target.value;
+    setPassword(newPassword);
+    validatePasswords(newPassword, confirmPassword);
+    validatePasswordComplexity(newPassword);
+  };
+
+  // Handles input to confirm password
+  const handleConfirmPasswordChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const newConfirmPassword = e.target.value;
+    setConfirmPassword(newConfirmPassword);
+    validatePasswords(password, newConfirmPassword);
+  };
+
+  // Checks if passwords match and sets error
+  const validatePasswords = (
+    password: string | null,
+    confirmPassword: string | null,
+  ) => {
+    if (password !== confirmPassword) {
+      setPasswordError('Passwords do not match.');
+    } else {
+      setPasswordError(''); // Clear error when passwords match
+    }
+  };
+
+  // Set password complexity error if requirements are not met
+  const validatePasswordComplexity = (password: string | null) => {
+    const hasLowerCase = /[a-z]/.test(password || '');
+    const hasNumber = /\d/.test(password || '');
+    const longEnough = (password || '').length >= 8;
+
+    if (password && hasLowerCase && hasNumber && longEnough) {
+      setPasswordComplexity(true);
+      setPasswordComplexityError(null); // Clear error if all conditions are met
+    } else if (password) {
+      setPasswordComplexity(false);
+      setPasswordComplexityError('Password must meet complexity requirements');
+    } else {
+      setPasswordComplexity(false);
+      setPasswordComplexityError(null); // Clear error if password is empty
+    }
+  };
+
+  const handleSignUp = async () => {
+    if (password) {
+      await signUp(email, password);
+    }
+  };
+
+  return (
+    <>
+      <input
+        name="email"
+        onChange={e => setEmail(e.target.value)}
+        value={email}
+        placeholder="Email"
+      />
+      {/* Email input*/}
+      <PasswordInput
+        value={password}
+        onChange={handlePasswordChange}
+        placeholder="Password"
+        isVisible={showPassword}
+        toggleVisibility={() => setShowPassword(!showPassword)}
+        name="password"
+      />
+      {/* Password input with toggle visibility */}
+      <PasswordInput
+        value={confirmPassword}
+        onChange={handleConfirmPasswordChange}
+        placeholder="Confirm Password"
+        isVisible={showConfirmPassword}
+        toggleVisibility={() => setShowConfirmPassword(!showConfirmPassword)}
+        name="confirmPassword"
+      />
+      {/* Confirm password input with toggle visibility */}
+      <button
+        type="button"
+        onClick={handleSignUp}
+        disabled={!email || !!passwordError || !!passwordComplexityError}
+      >
+        Sign up
+      </button>{' '}
+      {/* Sign up button */}
+      {confirmPassword && passwordError && (
+        <p style={{ color: 'red' }}>{passwordError}</p>
+      )}
+      {/* Conditional password validation error message */}
+      <PasswordComplexity
```

Comment:
while the "Passwords Do Not Match" error is in the correct place rn, to match designs, `PasswordComplexity` should be directly under the Password input (i.e. above Confirm Password)